ts/WoltLabSuite/Core/Acp/Ui/User/Action/Handler/Dialog/Ban.ts

```diff
@@ -0,0 +1,146 @@
+  private static instance: BanDialog;
+
+  private banCallback: () => void;
+  private userIDs: number[];
+  private submitElement: HTMLElement;
+  private neverExpiresCheckbox: HTMLInputElement;
+  private reasonInput: HTMLInputElement;
+  private userBanExpiresSettingsElement: HTMLElement;
+  private dialogContent: HTMLElement;
+  public static open(userIDs: number[], callback: () => void): void {
+    if (!BanDialog.instance) {
+      BanDialog.instance = new BanDialog();
+    }
+
+    BanDialog.instance.setCallback(callback);
+    BanDialog.instance.setUserIDs(userIDs);
+    BanDialog.instance.openDialog();
+  }
+
+  private openDialog(): void {
+    UiDialog.open(this);
+  }
+
+  private setCallback(callback: () => void): void {
+    this.banCallback = callback;
+  }
+
+  private setUserIDs(userIDs: number[]) {
+    this.userIDs = userIDs;
+  }
+
+  private banSubmit(reason: string, expires: string): void {
+    Ajax.apiOnce({
+      data: {
+        actionName: "ban",
+        className: "wcf\\data\\user\\UserAction",
+        objectIDs: this.userIDs,
+        parameters: {
+          banReason: reason,
+          banExpires: expires,
+        },
+      },
+      success: this.banCallback,
+    });
+  }
+
+  private cleanupDialog(): void {
+    this.reasonInput.value = "";
+    this.neverExpiresCheckbox.checked = true;
+    DatePicker.clear("userBanExpires");
+    this.userBanExpiresSettingsElement.style.setProperty("display", "none", "");
+  }
+
+  _dialogSetup(): ReturnType<DialogCallbackSetup> {
+    return {
+      id: "userBanHandler",
+      options: {
+        onSetup: (content: HTMLElement): void => {
+          this.dialogContent = content;
+          this.submitElement = content.querySelector(".formSubmitButton")! as HTMLElement;
+          this.reasonInput = content.querySelector("#userBanReason")! as HTMLInputElement;
+          this.neverExpiresCheckbox = content.querySelector("#userBanNeverExpires")! as HTMLInputElement;
+          this.userBanExpiresSettingsElement = content.querySelector("#userBanExpiresSettings")! as HTMLElement;
+
+          this.submitElement.addEventListener("click", (event) => {
+            event.preventDefault();
+
+            const expires = this.neverExpiresCheckbox.checked ? "" : DatePicker.getValue("userBanExpires");
+            this.banSubmit(this.reasonInput.value, expires);
+
+            UiDialog.close(this);
+
+            this.cleanupDialog();
+          });
+
+          this.neverExpiresCheckbox.addEventListener("change", (event) => {
+            const checkbox = event.currentTarget as HTMLInputElement;
+            if (checkbox.checked) {
+              this.userBanExpiresSettingsElement.style.setProperty("display", "none", "");
+            } else {
+              this.userBanExpiresSettingsElement.style.removeProperty("display");
+            }
+          });
+        },
+        title: Language.get("wcf.acp.user.ban.sure"),
+      },
+      source: `<div class="section">
```

Comment:
I would start the HTML code in the next line and start with no indentation reduce the overall indentation of the following lines.